docs/gettingstarted/structuring.md

```diff
@@ -1,124 +1,81 @@
-
-  [[dependencies.examplemod]]
-    modId="forge"
-    mandatory=true
-    versionRange="[45,)"
-    ordering="NONE"
-    side="BOTH"
-
-  [[dependencies.examplemod]]
-    modId="minecraft"
-    mandatory=true
-    versionRange="[1.19.4]"
-    ordering="NONE"
-    side="BOTH"
+When structuring your mod, pick a unique, top-level package structure. Many programmers will use the same name for different classes, interfaces, etc. Java allows classes to have the same name as long as they are in different packages. As such, if two classes have the same package with the same name, only one would be loaded, most likely causing the game to crash.
+
+```
+a.jar
+  - com.example.ExampleClass
+b.jar
+  - com.example.ExampleClass // This class will not normally be loaded
 ```
 
-If any string is specified as `${file.jarVersion}`, Forge will replace the string with the **Implementation Version** specified in your jar manifest at runtime. Since the user development environment has no jar manifest to pull from, it will be `NONE` instead. As such, it is usually recommended to leave the `version` field alone. Here is a table of attributes that may be given to a mod, where `mandatory` means there is no default and the absence of the property causes an error.
+This is even more relevant when it comes to loading modules. If there are class files in two packages under the same name in separate modules, this will cause the mod loader to crash on startup since mod modules are exported to the game and other mods.
 
-|     Property  |   Type   | Default           | Description |
-|-------------: |:--------:|:--------:         |:------------|
-|        modid  |  string  | mandatory         | The modid this file is linked to. |
-|      version  |  string  | mandatory         | The version of the mod. It should be just numbers separated by dots, ideally conforming to Forge's [Semantic Versioning][versioning] structure. |
-|  displayName  |  string  | mandatory         | The user-friendly name of this mod. |
-| updateJSONURL |  string  |   `""`            | The URL to a [version JSON][updatechecker]. |
-|   displayURL  |  string  |   `""`            | A link to the mod's homepage. |
-|     logoFile  |  string  |   `""`            | The filename of the mod's logo. It must be placed in the root resource folder, not in a subfolder. |
-|      credits  |  string  |   `""`            | A string that contains any acknowledgements you want to mention. |
-|      authors  |  string  |   `""`            | The authors of this mod. |
-|  description  |  string  | mandatory         | A description of this mod. |
-| displayTest   | string   | `"MATCH_VERSION"` | Controls the display of the mod in the server connection screen.
-| dependencies  | [list]   |   `[]`            | A list of dependencies of this mod. |
+```
+module A
+  - package X
+    - class I
+    - class J
+module B
+  - package X // This package will not normally be loaded, preventing usage of R, S, T
```

Comment:
As a bit of nuance: By my understanding, Java doesn't crash by the mere presence of two modules have the same package. The part that causes the crash is when two modules co-exist and try to export that same package out to another module (or all other modules, in the case of `open` modules).

As Java provides APIs to build module layers programmatically, an application or framework can load multiple modules that have the same package as long as each module does not co-exist with another on the same layer and export the same package.

Reply:
I'm not sure whether that should be stated, but I fixed the message to reflect the text above mentioning the crash.